Classes/common/HTTP/CDTHTTPInterceptor.h

```diff
@@ -0,0 +1,29 @@
+//
+//  CDTURLSessionInterceptor.h
+//  
+//
+//  Created by Rhys Short on 24/08/2015.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+
+#import "CDTHTTPInterceptorContext.h"
+
+@protocol CDTHTTPInterceptor <NSObject>
+
+/**
+ *  Intercepts a request before it is made
+ *
+ *  @param context the context for this interception
+ *
+ *  @return the context for this interception
+ **/
+@optional
```

Comment:
Should be above the doc comment. Remember this isn't an annotation of the method, see [here](https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/WorkingwithProtocols/WorkingwithProtocols.html#//apple_ref/doc/uid/TP40011210-CH11-SW6)